### DISCUSSION TITLE
Query: Django-Channels with Redis Channels layer

### DISCUSSION THREAD

**Discussion by yogimogi** [django/channels#1902] [General] [answered]
Hello,

This is our setup.
There is a Django app using Django channels. There are 2 instances of the app (say app1 and app2) running behind NGINX as load balancer. NGINX is set up to route traffic in round-robin fashion. There is a web application talking to Django apps via NGINX. User does something which in turn results in Web-socket being created. Say, the request for web-socket goes to **app1**. Now user does something else and that http request goes to **app2**, but with Django-channels with Redis as the channels layer, we are able to communicate back on the socket which was created with **app1**, though http request was served by **app2**. Clearly,  this doesn't work if we use In-Memory as channels layer. We were trying to understand how Redis channels layer make this magic happen? 

regards, Yogesh

**Comment by carltongibson** [comment, marked as answer]
When you use Redis, both instances are taking to the same channel layer, so they can communicate. 

With the in memory layer, each instance has its own layer, so they can't.
